src/ray/util/util.h

```diff
@@ -94,6 +101,12 @@ struct EnumClassHash {
   }
 };
 
+/// Uses sscanf() to read a token matching from the string, advancing the iterator.
+/// \param c_str A string iterator that is dereferenceable. (i.e.: c_str < string::end())
+/// \param format The pattern. It must not produce any output. (e.g., use %*d, not %d.)
+/// \return The scanned prefix of the string, if any.
+std::string ScanToken(std::string::const_iterator &c_str, std::string format);
```

Comment:
Let's make this function private. We also have a copy of it in `url.cc`, so maybe we can unify them (e.g. by putting everything into `util.cc`).

Reply:
Okay sure.